### PR TITLE
Add Pinned Posts

### PR DESCRIPTION
**Pull request by davidgrzyb**
This PR adds the pinned posts feature by adding an `is_pinned` column to `canvas_posts`. It also includes two new scopes on `Post` for getting pinned posts and getting all posts ordered by `is_pinned` and then by `published_at`.

image

### CODE REVIEW

**Comment by austintoddj** [comment]
Thanks for submitting this @davidgrzyb! First question, is this more or less primarily for consumption by a frontend implementation?

**Comment by davidgrzyb** [comment]
Hi, yes this just makes it easier to implement pinned posts by the frontend. My thinking was ‘Post::posts()->paginate()’ would be the simplest way for the frontend to get the posts it needs.

**Comment by austintoddj** [comment]
Makes sense. Ok, I'll dig into this in the next couple days. There are a number of changes to the backend, but we should be able to get this integrated. I'll let you know what changes need to happen.

File: tests/PostTest.php [hunk 1]
```diff
@@ -52,4 +52,29 @@ public function allow_posts_to_share_the_same_slug_with_unique_users()
             'user_id' => $post_2->decodeResponseJson()['user_id'],
         ]);
     }
+
```
Comment (rami0-4717) [inline tests/PostTest.php:55]:
```suggestion
rami0-4717
```

**Comment by austintoddj** [comment]
Thanks for submitting this @davidgrzyb, but I'm not planning on putting this into the package right now. Appreciate the PR though!

**Pull request closed without merging**